test/topology_custom/test_replace_with_encryption.py

```diff
@@ -7,42 +7,54 @@
 from test.pylib.manager_client import ManagerClient
 from test.pylib.scylla_cluster import ReplaceConfig
 
+
 @pytest.mark.asyncio
-async def test_replace_with_encryption(manager: ManagerClient):
+@pytest.mark.parametrize("join_ring", [True, False])
+async def test_replace_with_encryption(manager: ManagerClient, join_ring):
     """Test that a node can be replaced if inter-dc encryption is enabled.
        The test creates 6 node cluster with two DCs and replaces one node in
-       each DC"""
+       each DC. The test is parametrized to confirm behavior in zero token node cases as well."""
     config = {"endpoint_snitch": "GossipingPropertyFileSnitch"}
+    config_maybe_zero_token = {"endpoint_snitch": "GossipingPropertyFileSnitch", "join_ring": join_ring}
     property_file_dc1 = {"dc": "dc1", "rack": "myrack"}
     property_file_dc2 = {"dc": "dc2", "rack": "myrack"}
 
-    s1 = await manager.servers_add(3,
-            config=config,
-            property_file=property_file_dc1,
-            server_encryption="dc"
-          )
-    s2 = await manager.servers_add(3,
-            config=config,
-            property_file=property_file_dc2,
-            server_encryption="dc"
-          )
-
-    await manager.server_stop_gracefully(s1[1].server_id)
-
-    replace_cfg = ReplaceConfig(replaced_id = s1[1].server_id, reuse_ip_addr = False,
-                                use_host_id = True)
-
-    await manager.server_add(replace_cfg=replace_cfg, config=config,
+    await manager.servers_add(3,
+                              config=config,
+                              property_file=property_file_dc1,
+                              server_encryption="dc"
+                              )
+    await manager.servers_add(3,
+                              config=config,
+                              property_file=property_file_dc2,
+                              server_encryption="dc"
+                              )
+
+    zero_token_s1 = await manager.server_add(config=config_maybe_zero_token,
```

Comment:
`zero_token_s1` and  `zero_token_s2` are not the best names for nodes that don't have to be zero-token. I would leave `s1` and `s2` as the names.

Reply:
done